content/docs/iac/concepts/stash.md

```diff
@@ -0,0 +1,701 @@
+            ["Team"] = "platform",
+        },
+    },
+});
+```
+
+{{% /choosable %}}
+{{% choosable language java %}}
+
+```java
+var configStash = new Stash("configStash", StashArgs.builder()
+    .input(Map.of(
+        "region", "us-west-2",
+        "instanceType", "t3.micro",
+        "tags", Map.of(
+            "Environment", "production",
+            "Team", "platform"
+        )
+    ))
+    .build());
+```
+
+{{% /choosable %}}
+{{% choosable language yaml %}}
+
+```yaml
+resources:
+  configStash:
+    type: pulumi:pulumi:Stash
+    properties:
+      input:
+        region: us-west-2
+        instanceType: t3.micro
+        tags:
+          Environment: production
+          Team: platform
+```
+
+{{% /choosable %}}
+
+{{< /chooser >}}
+
+## Stashing secret values
+
+The `Stash` resource respects secret annotations. If the `input` value is marked as a secret, the `output` will also be secret, and the value will be encrypted in your stack's state.
+
+{{< chooser language "typescript,python,go,csharp,java,yaml" >}}
+
+{{% choosable language typescript %}}
+
+```typescript
+const apiKeyStash = new pulumi.Stash("apiKeyStash", {
+    input: pulumi.secret("my-secret-api-key"),
+});
+
+// The output is also marked as secret
+export const apiKey = apiKeyStash.output;
+```
+
+{{% /choosable %}}
+{{% choosable language python %}}
+
+```python
+api_key_stash = pulumi.Stash("apiKeyStash",
+    input=pulumi.Output.secret("my-secret-api-key"))
+
+# The output is also marked as secret
+pulumi.export("apiKey", api_key_stash.output)
+```
+
+{{% /choosable %}}
+{{% choosable language go %}}
+
+```go
+apiKeyStash, err := pulumi.NewStash(ctx, "apiKeyStash", &pulumi.StashArgs{
+    Input: pulumi.ToSecret(pulumi.String("my-secret-api-key")),
+})
+if err != nil {
+    return err
+}
+
+// The output is also marked as secret
+ctx.Export("apiKey", apiKeyStash.Output)
+```
+
+{{% /choosable %}}
+{{% choosable language csharp %}}
+
+```csharp
+var apiKeyStash = new Stash("apiKeyStash", new StashArgs
+{
+    Input = Output.CreateSecret("my-secret-api-key"),
+});
+
+// The output is also marked as secret
+return new Dictionary<string, object?>
+{
+    ["apiKey"] = apiKeyStash.Output,
+};
+```
+
+{{% /choosable %}}
+{{% choosable language java %}}
+
+```java
+var apiKeyStash = new Stash("apiKeyStash", StashArgs.builder()
+    .input(Output.secret("my-secret-api-key"))
+    .build());
+
+// The output is also marked as secret
+ctx.export("apiKey", apiKeyStash.output());
+```
+
+{{% /choosable %}}
+{{% choosable language yaml %}}
+
+```yaml
+resources:
+  apiKeyStash:
+    type: pulumi:pulumi:Stash
+    properties:
+      input:
+        fn::secret: my-secret-api-key
+
+outputs:
+  apiKey: ${apiKeyStash.output}
+```
+
+{{% /choosable %}}
+
+{{< /chooser >}}
+
+When viewing stashed secret values, their plaintext content will not be shown by default. Instead, they will be displayed as `[secret]` in the CLI. Pass `--show-secrets` to reveal the plaintext value.
```

Comment:
pass `--show-secrets` where? `up`/`preview`/etc.?

Reply:
up, preview, stack, export, there's a load of commands that take this option. Maybe 
"Pass `--show-secrets` to the command ran to reveal the plaintext value."?